app/controllers/network_router_controller.rb

```diff
@@ -12,4 +12,259 @@ class NetworkRouterController < ApplicationController
+      end
+    end
+    process_network_routers(routers_to_delete, "destroy") unless routers_to_delete.empty?
+
+    # refresh the list if applicable
+    if @lastaction == "show_list"
+      show_list
+      @refresh_partial = "layouts/gtl"
+    elsif @lastaction == "show" && @layout == "network_router"
+      @single_delete = true unless flash_errors?
+      if @flash_array.nil?
+        add_flash(_("The selected Router was deleted") % {:model => ui_lookup(:table => "network_router")})
+      end
+    end
+  end
+
+  def edit
+    assert_privileges("network_router_edit")
+    @router = find_by_id_filtered(NetworkRouter, params[:id])
+    @in_a_form = true
+    drop_breadcrumb(
+      :name => _("Edit Router \"%{name}\"") % {:model => ui_lookup(:table => 'network_router'), :name => @router.name},
+      :url  => "/network_router/edit/#{@router.id}"
+    )
+  end
+
+  def get_checked_router_id(params)
+    if params[:id]
+      checked_router_id = params[:id]
+    else
+      checked_routers = find_checked_items
+      checked_router_id = checked_routers[0] if checked_routers.length == 1
+    end
+    checked_router_id
+  end
+
+  def update
+    assert_privileges("network_router_edit")
+    @router = find_by_id_filtered(NetworkRouter, params[:id])
+    options = form_params
+    case params[:button]
+    when "cancel"
+      cancel_action(_("Edit of Router \"%{name}\" was cancelled by the user") % {
+        :model => ui_lookup(:table => 'network_router'),
+        :name  => @router.name
+      })
+
+    when "save"
+      task_id = @router.update_network_router_queue(session[:userid], options)
+
+      add_flash(_("Router update failed: Task start failed: ID [%{id}]") %
+                {:id => task_id.inspect}, :error) unless task_id.kind_of?(Fixnum)
+
+      if @flash_array
+        javascript_flash(:spinner_off => true)
+      else
+        initiate_wait_for_task(:task_id => task_id, :action => "update_finished")
+      end
+    end
+  end
+
+  def update_finished
+    task_id = session[:async][:params][:task_id]
+    router_id = session[:async][:params][:id]
+    router_name = session[:async][:params][:name]
+    task = MiqTask.find(task_id)
+    if MiqTask.status_ok?(task.status)
+      add_flash(_("%{model} \"%{name}\" updated") % { :model => ui_lookup(:table => 'network_router'),
+                                                      :name  => router_name })
+    else
+      add_flash(
+        _("Unable to update %{model} \"%{name}\": %{details}") % { :model   => ui_lookup(:table => 'network_router'),
+                                                                   :name    => router_name,
+                                                                   :details => task.message }, :error)
+    end
+
+    @breadcrumbs.pop if @breadcrumbs
+    session[:edit] = nil
+    session[:flash_msgs] = @flash_array.dup if @flash_array
+
+    javascript_redirect :action => "show", :id => router_id
+  end
+
+  private
+
+  def form_params
```

Comment:
Yeah, maybe, but again, I've been trying to stick with existing code, which doesn't necessarily makes it write but at least makes it consistent for future change.
To be honest with you it seems, at least to me, that the model design needs to reconsider things. This method shouldn't even exist at all, but at the moment we don't have a choice, we writting data on the cloud side directly, through raw commands because the model won't do that for us yet. So basically this is going away, hopefully and ultimately.